src/native/components/calendar/Calendar.js

```diff
@@ -116,6 +116,8 @@ class Calendar extends Component {
     // Empty Calendar
     if(this.dates.length === 0) return null;
 
+    this.dates = getDates(activeCatalogo, futureCatalogos);
```

Comment:
should this happen in `componentWillReceiveProps`?

Reply:
No, `activeCatalogo` and `futureCatalogos` come from props so a change in those values trigger rerender. The problem was that we were only calling `this.dates = getDates()` on the constructor. So when props changed `this.dates` wasn't updating.

Reply:
yh but couldn't we use that event rather then putting this logic here in the `render`?

Reply:
Is not what `componentWillReceiveProps` is for. Also is soon to be [depreacated](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops)